optimizely/event_builder.py

```diff
@@ -243,40 +243,39 @@ def _get_required_params_for_conversion(self, event_key, event_tags, decisions):
     Returns:
       Dict consisting of the decisions and events info for conversion event.
     """
-
+    snapshot = {}
     for experiment_id, variation_id in decisions:
-      snapshot = {}
+
       experiment = self.config.get_experiment_from_id(experiment_id)
 
       if variation_id:
-        snapshot[self.EventParams.DECISIONS] = [{
+        snapshot.setdefault(self.EventParams.DECISIONS, []).append({
           self.EventParams.EXPERIMENT_ID: experiment_id,
           self.EventParams.VARIATION_ID: variation_id,
           self.EventParams.CAMPAIGN_ID: experiment.layerId
-        }]
-
-        event_dict = {
-          self.EventParams.EVENT_ID: self.config.get_event(event_key).id,
-          self.EventParams.TIME: self._get_time(),
-          self.EventParams.KEY: event_key,
-          self.EventParams.UUID: str(uuid.uuid4())
-        }
+        })
 
-        if event_tags:
-          revenue_value = event_tag_utils.get_revenue_value(event_tags)
-          if revenue_value is not None:
-            event_dict[event_tag_utils.REVENUE_METRIC_TYPE] = revenue_value
+    event_dict = {
+      self.EventParams.EVENT_ID: self.config.get_event(event_key).id,
+      self.EventParams.TIME: self._get_time(),
+      self.EventParams.KEY: event_key,
+      self.EventParams.UUID: str(uuid.uuid4())
+    }
 
-          numeric_value = event_tag_utils.get_numeric_value(event_tags, self.config.logger)
-          if numeric_value is not None:
-            event_dict[event_tag_utils.NUMERIC_METRIC_TYPE] = numeric_value
+    if event_tags:
+      revenue_value = event_tag_utils.get_revenue_value(event_tags)
+      if revenue_value is not None:
+        event_dict[event_tag_utils.REVENUE_METRIC_TYPE] = revenue_value
 
-          if len(event_tags) > 0:
-            event_dict[self.EventParams.TAGS] = event_tags
+      numeric_value = event_tag_utils.get_numeric_value(event_tags, self.config.logger)
+      if numeric_value is not None:
+        event_dict[event_tag_utils.NUMERIC_METRIC_TYPE] = numeric_value
 
-        snapshot[self.EventParams.EVENTS] = [event_dict]
+      if len(event_tags) > 0:
+        event_dict[self.EventParams.TAGS] = event_tags
 
-        return snapshot
+    snapshot.setdefault(self.EventParams.EVENTS, []).append(event_dict)
```

Comment:
Do we need to setdefault as opposed to just initializing an empty array with that `event_dict`?

Reply:
Yeah, let me update that.